evm-adapters/src/lib.rs

```diff
@@ -52,6 +52,8 @@ pub trait Evm<State> {
     /// Gets the revert reason type
     fn revert() -> Self::ReturnReason;
 
+    fn expected_revert(&self) -> Option<Vec<u8>>;
```

Comment:
I'd return an `Option<&[u8]>` to avoid the clone in implementations

Reply:
OK. Rust n00b questions, why not `Option<&Vec<u8>>` ?

Reply:
Don't have a good answer, more idiomatic AFAIK. Maybe @mattsse